src/simple-cache-client.ts

```diff
@@ -195,9 +220,19 @@ export class SimpleCacheClient {
   public async listSigningKeys(
     nextToken?: string
   ): Promise<ListSigningKeysResponse> {
+    const client = this.getNextDataClient();
     return await this.controlClient.listSigningKeys(
-      this.dataClient.getEndpoint(),
+      client.getEndpoint(),
       nextToken
     );
   }
+
+  private getNextDataClient(): MomentoCache {
+    if (this.nextDataClientIndex >= this.dataClients.length) {
+      this.nextDataClientIndex = 0;
+    }
+    const client = this.dataClients[this.nextDataClientIndex];
+    this.nextDataClientIndex++;
+    return client;
```

Comment:
```suggestion
    const client = this.dataClients[this.nextDataClientIndex];
    this.nextDataClientIndex = (this.nextDataClientIndex + 1) % this.dataClients.length;
    return client;
```
nit: unnecessary condition